server/routes/users.js

```diff
@@ -424,30 +413,30 @@ router.post('/verify', async ctx => {
  * @apiParam (Required Params) {string} user[email] Unique email
  *
  */
-router.get('/verify', async ctx => {
+
+router.get('/verify', permController.requireAuth, async ctx => {
   const decodedMail = Buffer.from(ctx.query.email, 'base64').toString('ascii');
-  const user = await User.query().findOne({ 'email': decodedMail, 'resetPasswordToken': ctx.query.token });
-  ctx.assert(user, 'No email found');
-  let message;
+  let user = await User.query().findOne({ 'email': decodedMail });
+  ctx.assert(user, 404, 'No email found');
+  let verifiedData;
   try {
     if (new Date() < user.resetPasswordExpires) {
```

Comment:
throw an error if the request has expired. With current flow, the endpoint will return 200 ok status